skl2onnx/operator_converters/gradient_boosting.py

```diff
@@ -28,32 +28,39 @@ def convert_sklearn_gradient_boosting_classifier(scope, operator, container):
     attrs = get_default_tree_classifier_attribute_pairs()
     attrs['name'] = scope.get_unique_operator_name(op_type)
 
-    if op.n_classes_ == 2:
-        transform = 'LOGISTIC'
-        # class_prior_ was introduced in scikit-learn 0.21.
-        if hasattr(op.init_, 'class_prior_'):
-            base_values = op.init_.class_prior_
-            assert base_values.shape == (2, )
-        else:
-            base_values = [op.init_.prior]
-        if op.loss == 'deviance':
-            # See https://github.com/scikit-learn/scikit-learn/blob/
-            # master/sklearn/ensemble/_gb_losses.py#L666.
-            eps = np.finfo(np.float32).eps
-            base_values = np.clip(base_values, eps, 1 - eps)
-            base_values = np.log(base_values / (1 - base_values))
+    transform = 'LOGISTIC' if op.n_classes_ == 2 else 'SOFTMAX'
+    if op.init == 'zero':
+        base_values = np.zeros(op.loss_.K)
+    elif op.init is None:
+        if op.n_classes_ == 2:
+            # class_prior_ was introduced in scikit-learn 0.21.
+            if hasattr(op.init_, 'class_prior_'):
+                base_values = op.init_.class_prior_
+                assert base_values.shape == (2, )
+            else:
+                base_values = [op.init_.prior]
+            if op.loss == 'deviance':
+                # See https://github.com/scikit-learn/scikit-learn/blob/
+                # master/sklearn/ensemble/_gb_losses.py#L666.
+                eps = np.finfo(np.float32).eps
+                base_values = np.clip(base_values, eps, 1 - eps)
+                base_values = np.log(base_values / (1 - base_values))
+            else:
+                raise NotImplementedError(
+                    "Loss '{0}' is not supported yet. You "
+                    "may raise an issue at "
+                    "https://github.com/onnx/sklearn-onnx/issues.".format(
+                        op.loss))
         else:
-            raise NotImplementedError(
-                "Loss '{0}' is not supported yet. You "
-                "may raise an issue at "
-                "https://github.com/onnx/sklearn-onnx/issues.".format(op.loss))
+            # class_prior_ was introduced in scikit-learn 0.21.
+            if hasattr(op.init_, 'class_prior_'):
+                base_values = op.init_.class_prior_
+            else:
+                base_values = op.init_.priors
     else:
-        transform = 'SOFTMAX'
-        # class_prior_ was introduced in scikit-learn 0.21.
-        if hasattr(op.init_, 'class_prior_'):
-            base_values = op.init_.class_prior_
-        else:
-            base_values = op.init_.priors
+        raise NotImplementedError(
+            'Setting init to an estimator is not supported, you may raise an '
```

Comment:
If setting init is doable, probably you can add an issue by yourself and put the link the exception, is it better?

Reply:
It's not as straight-forward. If init is set to an estimator, we'll have to call it's predict(), which we could do but the result would be a set of output names(which will be evaluated by onnxruntime). However, TreeClassifier and TreeRegressor onnx ops expect a list of floats as base_values. This makes the implementation tricky.